src/gccrs/args.rs

```diff
@@ -62,7 +62,7 @@ fn format_output_filename(
 
     match crate_type {
         CrateType::Bin => output_file.push(&format!("{}{}", crate_name, extra_filename)),
-        CrateType::DyLib => output_file.push(&format!("lib{}.so{}", crate_name, extra_filename)),
+        CrateType::DyLib => output_file.push(&format!("lib{}{}.so", crate_name, extra_filename)),
```

Comment:
I think you should extract the suffix .so, .a to be variables or a configuration class of some kind since these will be platform dependant. I think this is fine for this PR but you can turn this comment into a github-issue

Reply:
I see this has been opened to track this: https://github.com/Rust-GCC/cargo-gccrs/issues/18 I can resolve this now